### PR TITLE
Add Recursor DNSSEC docs

### PR DESCRIPTION
**Pull request by pieterlexis**
Note, already contains docs on the non-merged #3910 code.

### CODE REVIEW

File: docs/markdown/recursor/dnssec.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+When `dnssec` is set to `process` the behaviour is similar to [`process-no-validate`](#process-no-validate).
+However, when the query has the AD-bit set, the recursor will try to validate the
+data and set the AD-bit in the response when the data is validated and send a
+SERVFAIL on a bogus answer.
+
+## `validate-log`
+In this mode, the recursor will attempt to validate all data it retrieves from
+authoritative servers, regardless of the client's DNSSEC desires, and will log the
+validation result. This mode can be used to determine the extra load and amount
+of possibly bogus answers before turning on full-blown validation.
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: Perhaps clarify that to the endusers this is exactly the same as `process`.

File: docs/markdown/recursor/dnssec.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+|    | `off` | `process-no-validate` | `process` | `validate-log` | `validate` |
+|:------------|:-------|:-------------|:-------------|:-------------|:-------------|
+|Perform validation| No | No | Only on +AD from client | Always (logs result) | Always |
+|SERVFAIL on bogus| No | No | Only on +AD from client | Only on +AD from client | Always |
+|AD in response on authenticated data| Never | Never | Only on +AD from client | Only on +AD from client | Only on +AD from client |
+|RRSIGs/NSECs in answer on +DO from client| No | Yes | Yes | Yes | Yes |
+
+# Trust Anchor Management
+The PowerDNS Recursor ships with the DNSSEC Root key built-in. However, all trust
+anchors are configurable.
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: Maybe carefully clarify current lack of 5011 here.

File: docs/markdown/recursor/dnssec.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+## Trust Anchors
+Configuring DNSSEC keys must be done in the [`lua-config-file`](settings.md#lua-config-file),
+using `addDS`. This function takes 2 arguments, the node in the DNS-tree and the
+data of the correspoding DS record. To e.g. add a trust anchor for the root and
+powerdn.com, use the following config in the Lua file:
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: powerdns.com

File: docs/markdown/recursor/dnssec.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+|RRSIGs/NSECs in answer on +DO from client| No | Yes | Yes | Yes | Yes |
+
+# Trust Anchor Management
+The PowerDNS Recursor ships with the DNSSEC Root key built-in. However, all trust
+anchors are configurable.
+
+## Trust Anchors
+Configuring DNSSEC keys must be done in the [`lua-config-file`](settings.md#lua-config-file),
+using `addDS`. This function takes 2 arguments, the node in the DNS-tree and the
+data of the correspoding DS record. To e.g. add a trust anchor for the root and
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: corresponding. Do you have an i/aspell button in your editor? I find it is very useful.

File: docs/markdown/recursor/dnssec.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+function requires the name of the zone and an optional reason:
+
+```lua
+addNTA('example.', "Someone messed up the delegation")
+addNTA('powerdns.com') -- No reason given
+```
+
+### Runtime Configuration of Negative Trust Anchors
+The [`rec_control`](running.md) command can be used to manage the negative trust
+anchors of a running instance.
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: Somewhere in this section, mention that these runtime configs do not persist.

File: docs/markdown/recursor/dnssec.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+Adding a negative trust anchor is done with the `add-nta` command(that optionally
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: whitespace before (

File: docs/markdown/recursor/dnssec.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+accepts a reason):
+
+```
+$ rec_control add-nta domain.example botched keyroll
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: Does this indeed work without quotes around `botched keyroll`?

File: docs/markdown/recursor/dnssec.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+To view the currently configured negative trust anchors, run `show-ntas`:
+
+```
+$ rec_control show-ntas
+Configured Negative Trust Anchors:
+subdomain.example.      Operator failed key-roll
+otherdomain.example.    DS in parent, no DNSKEY in zone
+```
+
+To remove a negative trust anchor(s), run `clear-nta`:
```
Comment (Habbie) [inline docs/markdown/recursor/dnssec.md]: grammar

**Comment by Habbie** [comment]
Just nits. Otherwise LGTM.

File: docs/markdown/recursor/dnssec.md (hunk 7 above)
Comment (pieterlexis) [inline docs/markdown/recursor/dnssec.md]: it does

**Comment by pieterlexis** [comment]
nits fixed and added positive trust anchor management information.

**Pull request merged**